storage/kv_store_rocksdb.py

```diff
@@ -101,10 +108,34 @@ def is_byte(self) -> bool:
         return True
 
     def has_key(self, key):
-        if isinstance(key, str):
-            key = key.encode()
+        key = self.to_byte_repr(key)
         return self._db.key_may_exist(key)[0]
 
     @property
     def closed(self):
         return self._db is None
+
+    @staticmethod
+    def _new_wrapped_iterator(itr, upper_bound):
+        # Takes Rocksdb iterator and an upper bound and returns another
+        # iterator which goes till the upper bound (inclusive)
+        return WrappingIter(itr, upper_bound)
+
+
+class WrappingIter:
+    def __init__(self, iterator, upper_bound):
+        self.iterator = iterator
+        self.upper_bound = upper_bound
+        self.reached_end = False
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        if self.reached_end is True:
+            raise StopIteration
+        item = next(self.iterator)
+        key = item[0] if isinstance(item, tuple) else item
```

Comment:
Why we do a special handling for tuples?

Reply:
Because the iterator `self.iterator` might be over over keys (`iterkeys`) or key and values both (`iteritems`)

Reply:
So, if this is a tuple, then we iterate over keys and values, and return a key only??